app/controllers/AdminController.php

```diff
@@ -0,0 +1,137 @@
+<?php
+//require_once 'models/PostModel.php';
+//require_once 'models/CommentModel.php';
+//require_once 'models/UserModel.php';
+namespace app\controllers;
+use app\models\CommentModel;
+use app\models\PostModel;
+use app\models\UserModel;
+use app\utils\View;
+
+class AdminController {
+
+    private PostModel $postModel;
+    private CommentModel $commentModel;
+    private UserModel $userModel;
+
+    public function __construct() {
+        $this->postModel = new PostModel();
+        $this->commentModel = new CommentModel();
+        $this->userModel = new UserModel();
+    }
+
+    public function index(): void
+    {
+        $posts = $this->postModel ->getPosts();
+//        require_once 'views/admin/dashboard.php';
+        View::render('views/admin/dashboard.php', ['posts' => $posts]);
+    }
+
+    public function show($id): void
+    {
+        $post = $this->postModel->getPostByID($id);
+        $comments = $this->commentModel->getApprovedComments($id);
+        if ($post == null) {
```

Comment:
@tony-tang-aligent I prefer the approach of initialising these arrays prior to using them.  I would suggest the first 2 lines of this method could be:

```
$posts = [];
$comments = [];
```